src/control/jobs/CustomExportJob.cpp

```diff
@@ -152,11 +151,11 @@ void CustomExportJob::run()
 		Document* doc = this->control->getDocument();
 
 		XojExportHandler h;
-		doc->lock();
-		h.prepareSave(doc);
-		h.saveTo(filename, this->control);
-		doc->unlock();
-
+		{
+			std::lock_guard<Document> _{*doc};
+			h.prepareSave(doc);
+			h.saveTo(filename, this->control);
+		}
```

Comment:
consider:   do a manual unlock instead of scope code block "side effect".

Reply:
Manual lock/unlock is error-prone and should be avoided. The scoped code block follows RAII and is idiomatic (i.e. the lock the automatically released when it goes out of scope.

Reply:
Mmh, you think someone could not understand the purpose of this block and wants to remove it. Forgetting about the effect of the std::lock_guard? On the other hand that's exactly the purpose of std::lock_guard.